Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -1300,19 +1354,32 @@ void MediaSource::sourceBufferActiveTrackFlagChanged(bool activeTrackFlag)
 
 void MediaSource::setMediaPlayerReadyState(MediaPlayer::ReadyState readyState)
 {
+    assertIsCurrent(m_dispatcher);
+
     if (!m_private)
         return;
     m_private->setMediaPlayerReadyState(readyState);
 }
 
 void MediaSource::memoryPressure()
 {
+    assertIsCurrent(m_dispatcher);
+
     if (!isManaged())
         return;
     for (auto& sourceBuffer : *m_sourceBuffers)
         sourceBuffer->memoryPressure();
 }
 
+void MediaSource::ensureOnDispatcher(Function<void()>&& function) const
+{
+    if (m_dispatcher->isCurrent()) {
+        function();
+        return;
+    }
+    m_dispatcher->dispatch(WTFMove(function));
+}
+
```

Comment:
I wonder if it would be useful for `ensureOnDispatcher()` to do the `weakThis`->`protectedThis` itself, and do the null check before calling the function. If you think this is valuable, you may need to rename it, something like `ensureProtectedOnDispatcher()` or `ensureWeakOnDispatcher()`.

Reply:
done